test/exhal/transcoder_test.exs

```diff
@@ -79,6 +79,23 @@ defmodule ExHal.TranscoderTest do
     assert {:ok, "http://example.com/3?data=INFO"} == ExHal.link_target(encoded, "fillin", tmpl_vars: [data: "INFO"])
   end
 
+  test "don't try to extract links from document that has no links" do
+    defmodule MyTinyTranscoder do
+      use ExHal.Transcoder
+
+      deflink "up", param: :mylink
+    end
+
+    hal = """
+    {
+      "_links": {}
+    }
+    """
```

Comment:
Please, add a test for `"up": []` and `"up": {"href": null}` too. Those are both legal, if awkward, formulations